### PR TITLE
Add disk space monitoring

### PR DESCRIPTION
**Pull request by evertiro**
Fixes issue https://github.com/Varying-Vagrant-Vagrants/VVV/issues/2380

Screen Shot 2021-03-15 at 2 17 06 AM

Screen Shot 2021-03-15 at 2 37 39 AM

### CODE REVIEW

File: dashboard-default.php [hunk 1]
```diff
@@ -30,6 +30,7 @@
 			<aside class="column right-column">
 				<?php
 				require_once __DIR__ . '/php/blocks/sidebar/inclusivity.php';
+				require_once __DIR__ . '/php/blocks/sidebar/disk-info.php';
```
Comment (tomjn) [inline dashboard-default.php:33]: I'd move this further down the sidebar

File: php/blocks/sidebar/disk-info.php [hunk 2]
```diff
@@ -0,0 +1,49 @@
+<div class="box disk-info">
+	<h3>Disk Info</h3>
```
Comment (tomjn) [inline php/blocks/sidebar/disk-info.php:2]:
```suggestion
	<h3>Disk space</h3>
```

File: php/blocks/sidebar/disk-info.php [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+	$v_free_space_gb = round( $v_free_space_mb / 1024, 1 );
+	$v_total_space   = round( disk_total_space( '/' ) / 1024 / 1024 / 1024, 0 );
+	$v_used_space    = round( $v_total_space - ( $v_free_space_mb / 1024 ), 1 );
+	$v_used_percent  = round( ( $v_used_space / $v_total_space ) * 100, 0 );
+	$v_low_space     = ( 200 > $v_free_space_mb );
+	$v_free_space    = ( $v_free_space_mb < 1024 ) ? $v_free_space_mb . ' MB' : $v_free_space_gb . ' GB';
+	?>
+	<div class="disk">
+		<div class="disk-header-meta">
+			<div class="disk-label">Virtual Machine</div>
```
Comment (tomjn) [inline php/blocks/sidebar/disk-info.php:15]:
```suggestion
			<div class="disk-label">VVV Guest Virtual Machine</div>
```

Lets eliminate as much ambiguity as possible

File: php/blocks/sidebar/disk-info.php [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+	$p_free_space_gb = round( $p_free_space_mb / 1024, 1 );
+	$p_total_space   = round( disk_total_space( '.' ) / 1024 / 1024 / 1024, 0 );
+	$p_used_space    = round( $p_total_space - ( $p_free_space_mb / 1024 ), 1 );
+	$p_used_percent  = round( ( $p_used_space / $p_total_space ) * 100, 0 );
+	$p_low_space     = ( 200 > $p_free_space_mb );
+	$p_free_space    = ( $p_free_space_mb < 1024 ) ? $p_free_space_mb . ' MB' : $p_free_space_gb . ' GB';
+	?>
+	<div class="disk">
+		<div class="disk-header-meta">
+			<div class="disk-label">Physical Machine</div>
```
Comment (tomjn) [inline php/blocks/sidebar/disk-info.php:36]:
```suggestion
			<div class="disk-label">Host Machine</div>
```

**Review by tomjn — CHANGES_REQUESTED** [review]
Looks great, there's some tweaking to do and in the future I'd like to separate out some of the styling to be more generic ( this isn't the only thing that I want to use a progress bar indicator for )

**Comment by Mte90** [comment]
My only question is how much is performant this integration for the rest seems good.

File: dashboard-default.php (hunk 1 above)
Comment (evertiro) [reply dashboard-default.php:33]: I put it at the top to mimic Plesk's positioning and so the visible indication would be obvious, but obviously it's movable wherever you'd like.

**Comment by evertiro** [comment]
> My only question is how much is performant this integration for the rest seems good.

It's not dynamic, only checking on page load. In my tests the difference it makes in load time is negligible.

**Comment by evertiro** [comment]
@tomjn I believe that covers all changes and brings it up to date

**Review by tomjn — APPROVED** [review]

**Pull request merged**